docs/DeveloperGuide.md

```diff
@@ -276,32 +276,40 @@ With the design considerations, we've chosen the alternatives that provide a bal
+4. This results in a `FilterCommand` object, which is executed by the `LogicManager`.
+5. The `FilterCommand` object can communicate with the `Model` when it is executed. It calls `Model#filterCompaniesByStatus(Predicate<Company> predicate)` to filter the list of companies by their application status.
+6. The result of the command execution is encapsulated as a `CommandResult` object which is returned back from `Logic`.
 
-The `filter` function is implemented in the `FilterCommand` class and uses the `FilterCommandParser` class to parse the
-arguments. The predicate class implementing `Predicate<Company>` is `ApplicationStatusPredicate`.
-- `ApplicationStatusPredicate` - Predicate to check if the company's application status is the same as the application
-  status specified in the command.
+The following sequence diagram will illustrate the process of performing the `filter` command.
+
+<img src="images/FilterSequenceDiagram.png"/>
+
+<div markdown="block" class="alert alert-info">
+**:information_source: Note:**
+The lifeline for `FilterCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+</div>
 
 #### 4.3.2 Design Considerations
 
-**Aspect: UI of the filter command**
+**Aspect: UI of the Filter Command**
 
-- **Alternative 1 (current choice):** The company details panel will be cleared whenever the filter command is executed.
-    - Pros: Users can focus on viewing details of company belonging to the filtered list only, reducing distractions and confusions.
-    - Cons: Users might have to execute the `view` command again to access details of the company that is selected even
-      if that company is in the filtered list, potentially leading to additional steps.
+* **Alternative 1:** The company details panel will still display the details of the company that was selected before the filter command is executed.
+  * Pros: Users can still view the details of the company in the company details panel alongside the filtered list of companies.
+  * Cons: Users may be confused as the currently viewed company in the company details panel may not be in the filtered list of companies.
 
-- **Alternative 2:** The company details panel will still display the details of the company that was selected before the
-  filter command is executed.
-    - Pros: Users can still view the details of the company in the company details panel alongside the filtered list of companies.
-    - Cons: Users may be confused as the currently viewed company in the company details panel may not be in the
-      filtered list of companies.
+* **Alternative 2 (current choice):** The company details panel will be cleared whenever the filter command is executed.
+  * Pros: Users can focus on viewing details of company belonging to the filtered list only, reducing distractions and confusions.
```

Comment:
users can focus on viewing details of "companies" or  "company(s)"  instead of "company"